### PR TITLE
Update LICENSE to BSD 2-clause

### PR DESCRIPTION
**Pull request by roversch**
@dkouzoup @giaf @zanellia @rienq @nielsvd @tmmsartor @FreyJo @bnovoselnik 

Please review :)

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,166 +1,252 @@
...
+    under certain conditions; type `show c' for details.
+
+The hypothetical commands `show w' and `show c' should show the appropriate parts of the General Public License. Of course, your program's commands might be different; for a GUI interface, you would use an “about box”.
+
+You should also get your employer (if you work as a programmer) or school, if any, to sign a “copyright disclaimer” for the program, if necessary. For more information on this, and how to apply and follow the GNU GPL, see <https://www.gnu.org/licenses/>.
+
+The GNU General Public License does not permit incorporating your program into proprietary programs. If your program is a subroutine library, you may consider it more useful to permit linking proprietary applications with the library. If this is what you want to do, use the GNU Lesser General Public License instead of this License. But first, please read <https://www.gnu.org/licenses/why-not-lgpl.html>.
+
+
+"CLASSPATH" EXCEPTION TO THE GPL
```
Comment (FreyJo) [inline LICENSE]: We should mark every file, the classpath exception applies to by adding the following sentence to the header:

> The authors designate this particular file as subject to the "Classpath"
> exception as provided by the authors in the LICENSE file that accompanied this code.

I think the files we should mark are all file in `acados` and in `interfaces/acados_c`.
Do you agree?

**Comment by giaf** [comment]
Before making any change to the code, it should be discussed once more the matter with Moritz in person.

I asked a meeting for that, so please do NOT merge the PR until that happened, thanks.

**Comment by nielsvd** [comment]
As a reminder, the COPYING file and the src/header files (at least the ones I checked) are still stating LGPL.

**Comment by doanminhdang** [comment]
I support this move to the permissive BSD license. We could expect more users from the industry and the project could get more resource to sustain.

**Comment by giaf** [comment]
Good for me. When @tmmsartor comes at the office, he will make some bash magic to change the license in the single .c/.h files.

**Comment by tmmsartor** [comment]
I am actually using a ready to use script to do that,
questions before pushing all the changes:
1- Should I wrap the LICENCE text to meet some reasonable column limit? 
2- Should I include the LICENCE text also on Makefile and CMake files?

**Comment by roversch** [comment]
Hi Tommaso, thanks for this effort.

1. I think that's best, yes. 80 or 100 columns should do
2. In my opinion, yes.
3. I would keep it.

**Pull request merged**